feat: add collapse to the table of content items

ahkohd opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### Description
<!-- Translators, try to follow this pattern when naming your PRs:
"i18n(language code): (short description)" -->
- What does this PR change? Give us a brief description. <!-- If it's an update try adding the commits as reference -->
- Closes # <!-- Add an issue number if this PR will close it or remove it. -->

<!--
Here's what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don't hesitate to ask any questions if you're not sure what these mean!

2. In a few minutes, you'll be able to see a preview of your changes on Netlify 🥳

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don't worry if this takes a day or two.

4. Reach out to us on Discord with any questions along the way:
   https://discord.com/invite/tauri
-->

ahkohd committed:
feat: add collapse to the table of content items

lucasfernog commented:
can we automatically collapse above the max heading level (instead of removing them from the list)? would make this feature neat

simonhyll reviewed (approved):
This is definitely an improvement! 🥳 We might wanna consider adding an option to the frontmatter of pages so that we can configure a `tableOfContents.collapseLevel` on a page by page basis, because I think on most pages it'll work great with default collapsing above the second level, but on e.g. the config reference pages we'd likely want to collapse already at the top level, at least in my opinion

But that's a bunch of extra effort, so I'm totally fine with merging as-is, since it's effectively currently not changing the appearance, it just adds a cool feature

Good job! 🙂

ahkohd committed:
feat: add a prop to automatically collapse at depth above a collapse level

By default depth level of 1

ahkohd committed:
feat: add collapse level to the table of contents schema

of the docs schema

ahkohd committed:
feat: add collapseLevel to toc prop

ahkohd committed:
feat: add collpaseLevel to the frontmatter

This is to test if the `collapseLevel` works, that it can be configured
through the document's frontmatter

ahkohd commented:
> This is definitely an improvement! 🥳 We might wanna consider adding an option to the frontmatter of pages so that we can configure a `tableOfContents.collapseLevel` on a page by page basis, because I think on most pages it'll work great with default collapsing above the second level, but on e.g. the config reference pages we'd likely want to collapse already at the top level, at least in my opinion
> 
> But that's a bunch of extra effort, so I'm totally fine with merging as-is, since it's effectively currently not changing the appearance, it just adds a cool feature
> 
> Good job! 🙂

Good day,

I have added support for configuring the collapse level via the document's frontmatter.

By default the collapse level is set to 1, counting starts from 0.

Commit 4476a67b40c36232b462fbff38b44afbfe86e3e3 shows an example usage.

ahkohd committed:
refactor: move to schema dir

ahkohd committed:
refactor: move docsSchema to schemas dir
